middleware.js

```diff
@@ -28,7 +61,7 @@ function webpackHotMiddleware(compiler, opts) {
     if (latestStats) {
       // Explicitly not passing in `log` fn as we don't want to log again on
       // the server
-      publishStats("sync", latestStats, eventStream);
+      publishStats("sync", latestStats, eventStream, null, res);
```

Comment:
passing the response object here seems a bit convoluted to me.

How about making `eventStream.hander` return the new client's ID, and passing that down instead?